dev/MANUALLY_GENERATING_IMAGE_CACHE_AND_CONSTRAINTS.md

```diff
@@ -424,7 +424,7 @@ The slight risk is that if there is a constraint problem that impacts regular PR
 make all PRs "red" until the constraint is fixed. However, if this is the case then usually we should fix
 the problem by fixing the tests or dependencies and the automated CI process should be able to self-heal.
 The main build does not use constraints and it will attempt to upgrade (or downgrade) the dependencies to
-the latest version matching the dependency specification we have in setup.cfg/setup.py/provider.yaml files.
+the latest version matching the dependency specification we have in pyproject.toml files.
```

Comment:
Isn’t the provider.yaml file still relevant?

Reply:
Yeah. `Provider.yaml` files are `single source of truth` (but pyproject.toml part of the deps is generated from them via pre-commit). I added explanation for that.